### PR TITLE
v4.1.x upgrade

### PR DESCRIPTION
**Pull request by hoank101**
- Upgrade cosmos-sdk from v0.46.15 -> v0.47.7
- Upgrade ibc v6 -> v7, use original IBC
- Upgrade wasmd v0.30.0 -> v0.45.0, use original wasmd
- Upgrade tokenfactory use terra version
- Add feeburn module
     +) Default burn 0%
     +) Add gov proposal 
- Moves the vested and reward token from the ContinuousVestingAccount of Notional Team to new multisig account. 

https://github.com/White-Whale-Defi-Platform/migaloo-chain/issues/297

### CODE REVIEW

**Review by duvbell — APPROVED** [review]
LGTM

File: Dockerfile [hunk 1]
```diff
@@ -14,11 +14,13 @@ RUN apk add --no-cache ca-certificates build-base git
...
-# See https://github.com/CosmWasm/wasmvm/releases
-ADD https://github.com/CosmWasm/wasmvm/releases/download/v1.2.1/libwasmvm_muslc.aarch64.a /lib/libwasmvm_muslc.aarch64.a
-ADD https://github.com/CosmWasm/wasmvm/releases/download/v1.2.1/libwasmvm_muslc.x86_64.a /lib/libwasmvm_muslc.x86_64.a
-RUN sha256sum /lib/libwasmvm_muslc.aarch64.a | grep 86bc5fdc0f01201481c36e17cd3dfed6e9650d22e1c5c8983a5b78c231789ee0
-RUN sha256sum /lib/libwasmvm_muslc.x86_64.a | grep a00700aa19f5bfe0f46290ddf69bf51eb03a6dfcd88b905e1081af2e42dbbafc
+# See https://github.com/CosmWasm/wasmvm/releases 
+ADD https://github.com/CosmWasm/wasmvm/releases/download/v1.5.0/libwasmvm_muslc.aarch64.a /lib/libwasmvm_muslc.aarch64.a
+ADD https://github.com/CosmWasm/wasmvm/releases/download/v1.5.0/libwasmvm_muslc.x86_64.a /lib/libwasmvm_muslc.x86_64.a
+
+# TODO: add checksums later
```
Comment (Sen-Com) [inline Dockerfile]: Here's an open TODO.

**Review by Sen-Com — COMMENTED** [review]
Great work. This is a significant improvement of our codebase. I particularly like the many tests.

File: Dockerfile (hunk 1 above)
Comment (duvbell) [reply Dockerfile]: oh. Lets me resolve that!

**Comment by hoank101** [comment]
@faddat please review

**Comment by faddat** [comment]
Hey guys, this looks really good, we only need to finish up with the linters, and that one other CI job that is failing.

**Review by faddat — APPROVED** [review]

**Review by faddat — APPROVED** [review]

**Pull request merged**